application.js

```diff
@@ -0,0 +1,278 @@
+        }
+    }
+}
+// 완료한 일 필터링
+function filterCompletedViewChange(event){
+    selectedFilter = filterCompleted
+    for(i=0;i<filterContainer.length;i++) {
+        filterContainer[i].classList.remove('selected')
+    }
+    filterCompleted.classList.add('selected')
+    
+    for(index=0;index<todoListCount;index++){
+        if(todoElements[index].querySelector('div input').getAttribute('checked') == null){
+            todoElements[index].style.display = "none"
+        } else {
+            todoElements[index].style.display = ""
+        }
+    }
+}
+
+// 할 일 삭제 이벤트 처리기.
+function removeCurrentTodoElement(event){
+    // 삭제된 할 일을 할 일 목록에서 제거.
+    removedTodoName = event.target.parentNode.querySelector('label').innerText
+    todoElementsNameArray.splice(todoElementsNameArray.indexOf(removedTodoName), 1)
+    // localStorage에 저장된 할 일 데이터를 지우고 할 일 목록을 업데이트.
+    ToDoElementStorage.removeItem(removedTodoName)
+    ToDoElementStorage.setItem(KEYWORD, JSON.stringify(todoElementsNameArray))
+    // 실제로 HTML 요소를 삭제하고 카운터 업데이트.
+    event.target.parentNode.parentNode.remove()
+    updateCountText(-1)
+}
+
+// 할 일 변경 이벤트 처리기.
+function updateTodoEdit(event){
+    todoElementLI = event.target.parentNode
+    // ESC를 눌렀다면 편집 모드 종료, Enter를 눌렀다면 편집 적용.
+    if(event.keyCode == ESC_KEYCODE){
+        todoElementLI.classList.toggle('editing')
+    } else if (event.keyCode == ENTER_KEYCODE){
+        // 각각 변경된 할 일 텍스트, 원래 할 일 텍스트.
+        newTodoText = event.target.value.trimStart().trimEnd()
+        updatedTodoText = todoElementLI.querySelector('div label').innerText
+        
+        // 입력값 필터링.
+        if(newTodoText.length == 0){
+            event.target.focus()
+        }
+        is_dup = false
+        todoElementsNameArray.forEach(element => {
+            if(element == newTodoText){
+                is_dup = true
+            }
+        })
+        if(is_dup){
+            alert('That ToDo is already exist!')
+            return;
+        }
+
+        // localStorage에서 해당 할 일의 완료 여부를 가져와서 따로 저장 후 해당 할 일 삭제.
+        status = ToDoElementStorage.getItem(updatedTodoText)
+        ToDoElementStorage.removeItem(updatedTodoText)
+        // localStorage에 변경된 할 일의 텍스트로 새로운 항목 저장. 위에서 따로 저장해둔 완료 여부를 적용.
+        ToDoElementStorage.setItem(newTodoText, status)
+        // 할 일 목록에서도 변경사항 적용 및 localStorage에 반영.
+        todoElementsNameArray.splice(todoElementsNameArray.indexOf(updatedTodoText), 1, newTodoText)
+        ToDoElementStorage.setItem(KEYWORD, JSON.stringify(todoElementsNameArray))
+        // HTML 요소에서도 변경사항 적용.
+        todoElementLI.querySelector('div label').innerText = newTodoText
+        todoElementLI.classList.toggle('editing')
+    }
+}
+
+function updateView(){
+    // ??? 구현 예정.
+}
+
+// 할 일이 몇 개 있는지 출력하는 텍스트(총 n 개)를 업데이트하는 로직.
+function updateCountText(change=0){
+    todoListCount += change
+    todoListCountText = document.querySelector('span.todo-count strong')
+    todoListCountText.innerText = todoListCount
+}
```

Comment:
```suggestion
function updateCountText(){
    todoListCountText = document.querySelector('span.todo-count strong');
    todoListCountText.innerText = todoList.querySelectorAll('li').length;
}
```
코드를 실행시켜보니 모든 항목을 보여줄 때는 count가 제대로 되는데 해야할 일, 완료할 일을 선택했을 때는 count가 제대로 되지 않네요. 코드를 이런식으로 수정하면 해야할 일, 완료할 일을 선택했을 때도 정상적으로 count가 될거 같습니다!

Reply:
카테고리를 선택했을 때도 '총 n 개' 문자열이 업데이트되야 한다는 건 생각하지 못했던 부분이었네요. 감사합니다!